torchao/quantization/quant_api.py

```diff
@@ -488,6 +488,24 @@ def int8_dynamic_activation_int8_semi_sparse_weight():
     """
     return int8_dynamic_activation_int8_weight(layout_type=SemiSparseLayoutType())
 
+def float8_weight_only():
+    """
+    Applies float8 weight-only symmetric per-channel quantization to linear layers.
+    """
+    def apply_float8wo_quant(weight):
+        # avoid circular dep
+        from torchao.dtypes import to_affine_quantized_float8
+
+        mapping_type = MappingType.SYMMETRIC
+        target_dtype = torch.float8_e4m3fn
+        eps = torch.finfo(torch.float32).eps
+        zero_point_dtype = torch.float32
+        block_size = (1, weight.shape[1])
+        return to_affine_quantized(input_float=weight, mapping_type=mapping_type, block_size=block_size, target_dtype=target_dtype,
```

Comment:
Should this be: to_affine_quantized_float8?

Reply:
we want to merge float8 into `to_affine_quantized` I think

Reply:
Merged from_float_float8 into from_float